### PR TITLE
Fixes #645: Extension to test for valid qualifier declaration constructor parameters

### LINKED ISSUE #645: qualifierdecl to cimxml fails
The following qualifierdecl:

        scopes = self._build_scope(set_true=['CLASS'])
        qd = CIMQualifierDeclaration('FooQualDecl', 'string', is_array=True,
                                     value='Some string',
                                     array_size=4, scopes=scopes,
                                     overridable=False, tosubclass=True,
                                     toinstance=True, translatable=False)

Fails the SetQualifier cim_operation request with the following error.
Traceback (most recent call last):
  File "testsuite/run_cim_operations.py", line 3477, in test_create_delete
    self.cimcall(self.conn.SetQualifier, qd)
  File "testsuite/run_cim_operations.py", line 153, in cimcall
    result = fn(*pargs, **kwargs)
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_operations.py", line 7830, in SetQualifier
    **extra)
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_operations.py", line 880, in _imethodcall
    for x in params.items() if x[1] is not None]
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_obj.py", line 4060, in tocimxml
    return value.tocimxml()
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_obj.py", line 3969, in tocimxml
    translatable=self.translatable)
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_xml.py", line 346, in __init__
    xval = VALUE_ARRAY(value)
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_xml.py", line 419, in __init__
    self.appendChildren(values)
  File "/home/kschopmeyer/.virtualenvs/pywbem27/local/lib/python2.7/site-packages/pywbem-0.11.0.dev0-py2.7.egg/pywbem/cim_xml.py", line 197, in appendChildren
    self.appendChild(child)
  File "/usr/lib/python2.7/xml/dom/minidom.py", line 113, in appendChild
    if node.nodeType == self.DOCUMENT_FRAGMENT_NODE:
AttributeError: 'unicode' object has no attribute 'nodeType'

The problem is the array=True which conflicts with the value.  The constructor does not compare the
types and error return but depends on cimxml which generates an error when constructing the xml.

### PR DESCRIPTION
**Pull request by KSchopmeyer**
Adds consistency tests for the multiple constructor paramters (is_array vs array_size and is_array vs. value is a list).

Add comment to changes.rst

### CODE REVIEW

File: pywbem/cim_obj.py [hunk 1]
```diff
@@ -3880,6 +3880,17 @@ def __init__(self, name, type, value=None, is_array=False,
         self.tosubclass = tosubclass
         self.toinstance = toinstance
         self.translatable = translatable
+        if is_array is None:
+            raise TypeError('None not allowed for is_array')
```
Comment (andy-maier) [inline pywbem/cim_obj.py]: IMO, this should raise a ValueError, instead.

File: pywbem/cim_obj.py [hunk 2]
```diff
@@ -3880,6 +3880,17 @@ def __init__(self, name, type, value=None, is_array=False,
...
+        if array_size and not is_array:
+            raise TypeError('QualifierDeclaration array_size allowed only '
```
Comment (andy-maier) [inline pywbem/cim_obj.py]: IMO, this should raise a ValueError, instead.

File: pywbem/cim_obj.py [hunk 3]
```diff
@@ -3880,6 +3880,17 @@ def __init__(self, name, type, value=None, is_array=False,
...
+                            'when is_array True.')
+        if isinstance(value, list) and not is_array:
+            raise TypeError('QualifierDeclaration is_array False and '
```
Comment (andy-maier) [inline pywbem/cim_obj.py]: IMO, this should raise a ValueError, instead.

File: pywbem/cim_obj.py [hunk 4]
```diff
@@ -3880,6 +3880,17 @@ def __init__(self, name, type, value=None, is_array=False,
...
+                            'value is a list.')
+        elif is_array and value is not None and not isinstance(value, list):
+            raise TypeError('QualifierDeclaration is_arrayue with scalar '
+                            'value')
```
Comment (andy-maier) [inline pywbem/cim_obj.py]: Fix typo in exception message.

**Comment by andy-maier** [comment]
-1, see comments.

File: pywbem/cim_obj.py (hunk 4 above)
Comment (KSchopmeyer) [inline pywbem/cim_obj.py]: Done

File: pywbem/cim_obj.py (hunk 3 above)
Comment (KSchopmeyer) [inline pywbem/cim_obj.py]: Done for all 4.  In summary all should be ValueError rather than TypeError

Comment (KSchopmeyer) [inline pywbem/cim_obj.py]: Think the difference is that you and I view Value vs Type a little differently. All are changed as is the test_cim_obj.py

**Comment by andy-maier** [comment]
Karl, thanks for the update.
Please rebase and resolve conflicts.

**Comment by KSchopmeyer** [comment]
+1.  Andy, Could you restart the AppVeyor build.  Failed somewhere but I cannot find a restart button.

**Comment by andy-maier** [comment]
+1

**Pull request merged**